### DISCUSSION TITLE
Allow to disable web notifications (noticed_notifications)

### DISCUSSION THREAD

**Discussion by javierav** [excid3/noticed#445] [General]
Hi @excid3!

I want to use the v2 version in my project but I have found a limitation: you cannot disable the insertion of the "web notification" (old delivery_by :database) into the database.

I'll give you an example. When I perform a certain action I want to notify the user by slack, email and mobile but I don't want the notification to appear in the list of web notifications.

I understand that to make these notifications the record is inserted into the noticed_events table, but why is it mandatory that it also be inserted into the noticed_notifications table? Shouldn't it be optional and user configurable?

In the current form, it is always required to remain as a web notification.

Regards!

**Comment by excid3** [comment]
Already supported with the `Noticed::Ephemeral` class. We just don't have docs on it yet.

**Reply by javierav** [reply]
Interesting! I'll try it then! Thanks!

**Reply by excid3** [reply]
Let me know how it goes. Will try and get our docs updated for that soon. I thought it was already in there somewhere.

**Comment by javierav** [comment]
I have started testing `Ephemeral` and have detected that in the context of an instance of `Noticed::Ephemeral`, `record` is `nil`. On the other hand, in `Noticed::Event` it does have the appropriate value. Is it for some reason?

**Reply by excid3** [reply]
That's the whole point, it doesn't save anything to the database so there is no `record`.

**Reply by javierav** [reply]
I thought that record was the record attribute (`Post.first`) that was passed to the notifier when calling

```
Post::CreateNotifier.with(record: Post.first).deliver(User.first)
```

**Reply by excid3** [reply]
I think you need to access it through params because that's a special case for the Event notifications to save in the polymorphic association.

**Reply by excid3** [reply]
We might be able to add a `delegate :record, to: :params` in Ephemeral to make that a bit more similar, but they are not guaranteed to be 100% compatible since they are different implementations.

**Reply by excid3** [reply]
Give https://github.com/excid3/noticed/pull/448 a try.

**Reply by javierav** [reply]
Yeah, thats works! Thank you for your involvement!

**Reply by excid3** [reply]
Great! Thanks for finding that. I totally missed that.
